File: android/app/src/main/java/jagtheora/ogg/OggSyncState.java
package jagtheora.ogg;

import jagtheora.misc.SimplePeer;

public final class OggSyncState extends SimplePeer {

    public OggSyncState() {
        this.init();
        if (this.b()) {
            throw new IllegalStateException();
        }
    }

    protected void clear() {}

    protected void init() {}

    public long pageSeek(OggPage arg0) { return 0L; }

    public int pageOut(OggPage arg0) { return 0; }

    public boolean write(byte[] arg0, int arg1) { return false; }

    public boolean reset() { return false; }
}

File: src/commonMain/kotlin/jagtheora/ogg/OggSyncState.kt
package jagtheora.ogg

import jagtheora.misc.SimplePeer

expect class OggSyncState : SimplePeer {
    fun reset(): Boolean

    override fun clear()

    fun write(arg0: ByteArray?, arg1: Int): Boolean

    fun pageOut(arg0: OggPage?): Int

    fun pageSeek(arg0: OggPage?): Long

    companion object {
        /*protected*/ fun init()
    }
}
